recipes/stx/all/conanfile.py

```diff
@@ -0,0 +1,147 @@
+        'shared': False,
+        'fPIC': True,
+    }
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.panic_backtrace == None:
+            self.options.panic_backtrace = (
+                self.options.backtrace and
+                not self.options.override_panic_handler
+            )
+
+        if self.options.visible_panic_hook == None:
+            self.options.visible_panic_hook = self.options.shared
+
+        compiler = self.settings.compiler
+        compiler_version = tools.Version(self.settings.compiler.version)
+        standards = ['17', '20', 'gnu17', 'gnu20']
+
+        if compiler.cppstd and not str(compiler.cppstd) in standards:
+            raise ConanInvalidConfiguration('STX requires C++17 support')
+
+        if compiler == 'Visual Studio' and compiler_version < 16:
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 support, use at least VS 2019'
+            )
+
+        if compiler == 'gcc' and compiler_version < 8:
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 support, use at least GCC 8'
+            )
+
+        if (compiler == 'clang' and compiler.libcxx and
+                compiler.libcxx in ['libstdc++', 'libstdc++11'] and
+                compiler_version < 9):
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which clang & libc++ < 10 lack'
+            )
+
+        if (compiler == 'clang' and compiler.libcxx and
+                compiler.libcxx == 'libc++' and
+                compiler_version < 10):
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which clang & libc++ < 10 lack'
+            )
+
+        if compiler == 'apple-clang':
+            raise ConanInvalidConfiguration(
+                'STX requires C++17 language and standard library features '
+                'which apple-clang and libc++ lack'
+            )
+
+        if (compiler == 'Visual Studio' and self.options.shared and
+                tools.Version(self.version) <= '1.0.1'):
+            raise ConanInvalidConfiguration(
+                'shared library build does not work on windows with '
+                'STX version <= 1.0.1'
+            )
+
+    def requirements(self):
+        if self.options.backtrace:
+            self.requires('abseil/20200923.1')
+
+    def source(self):
+        tools.get(**self.conan_data['sources'][self.version])
+        tools.rename(src=f'STX-{self.version}', dst='source_subfolder')
+        if self.version in self.conan_data['patches']:
+            for patch in self.conan_data['patches'][self.version]:
+                tools.patch(base_path='source_subfolder', **patch)
```

Comment:
This could be simplified,
You should add the base_path in the conandata.yml like https://github.com/conan-io/conan-center-index/pull/3178/files#diff-4a407dbdc3502e57c3223ab26114e6a4ff0a2a871eb649d897e76e7fa8d0b4ffR8
and then you can do
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
```

Reply:
Thanks!